src/odemis/driver/hamamatsurx_simcam.py

```diff
@@ -0,0 +1,552 @@
+          triggered, the DataFlow will start a new acquisition.
+        Behaviour is unspecified if the acquisition is already running.  # TODO still True?
+        event (model.Event or None): event to synchronize with. Use None to
+          disable synchronization.
+        The DataFlow can be synchronized only with one Event at a time.
+        """
+        # if event None and sync as well -> return, or if event sync, but sync already set -> return
+        if self._sync_event == event:
+            return
+
+        if self._sync_event:  # if new event = None, unsubscribe previous event (which was softwareTrigger)
+            self._sync_event.unsubscribe(self)
+            if not event:
+                self._evtq.put(None)  # in case it was waiting for this event
+
+        self._sync_event = event
+
+        if self._sync_event:
+            # softwareTrigger subscribes to onEvent method: if softwareTrigger.notify() called, onEvent method called
+            self._sync_event.subscribe(self)  # must have onEvent method
+            self._evtq = Queue.Queue()  # to be sure it's empty
+
+    @oneway
+    def onEvent(self):
+        """Called by the Event when it is triggered  (e.g. self.softwareTrigger.notify())."""
+        logging.debug("Event triggered to start a new synchronized acquisition.")
+        self._evtq.put(time.time())
+
+    # override
+    def updateMetadata(self, md):
+        super(ReadoutCamera, self).updateMetadata(md)
+        if model.MD_LENS_MAG in md:
+            self._updateWavelengthList()
+
+    def _mergeMetadata(self, md):
+        """Create dict containing all metadata from the children readout camera, streak unit, delay genereator
+        and the metadata from the parent streak camera."""
+
+        md_devices = [self.parent._streakunit._metadata, self.parent._delaybox._metadata]
+
+        for md_dev in md_devices:
+            for key in md_dev.keys():
+                if key not in md.keys():
+                    md[key] = md_dev[key]
+                else:
+                    md[key] = md[key] + ", " + md_dev[key]  # TODO change in real driver
+                    # md[key].append(md_dev[key])  # TODO make nice  ", ".join(c)
+        return md
+
+    def terminate(self):
+        try:
+            self._stop()  # stop any acquisition
+        except Exception:  # TODO which exception?
+            pass
+
+    def _start(self):
+        if self._generator is not None:
+            logging.warning("Generator already running")
+            return
+        self._generator = util.RepeatingTimer(self.exposureTime.value,
+                                              self._generate,
+                                              "SimCam image generator")
+        self._generator.start()
+
+    def _stop(self):
+        if self._generator is not None:
+            self._generator.cancel()
+            self._generator = None
+
+    def _getNewImage(self):
```

Comment:
Put all the "magic" of `img_counter` inside here, so the rest of the code doesn't need to care.
(IOW, do the self._img_counter += 1 here too).
I'd suggest to write something like:
`self._img_counter = (self._img_counter + 1) % len(self._img)`